File: Searching/Binary Search/Python/BinarySearch.py
def binarySearch(arr, n, key):
    low = 0
    high = n-1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == key:
            return mid
        elif arr[mid] > key:
            high = mid - 1
        elif arr[mid] < key:
            low = mid + 1
    
    return -1

arr = [1, 4, 6, 10, 20, 39, 50, 66]
n = len(arr)
key = 39
# calling function binary search
result = binarySearch(arr, n, key)
if result == -1:
    print("Element is not present in the array.")
else:
    print("Element is found at index: ", result)

File: Searching/Binary Search/C++/binarySearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int binarySearch(vector<int>nums, int target){
     int right = (nums.size())-1;
     int left = 0;
     int middle;
     while(right >= left)
     {
         middle = left + (right - left) / 2;
         if(nums[middle] == target){
            return middle;
         }
         else if(nums[middle] > target){
            right = (middle-1);
         }
         else{
            left = (middle+1);
         }
    }
    return -1;
}

int main()
{
    unsigned int i;
    int n;
    vector<int>nums;
    cout << "Enter the amount of numbers to be evaluated: ";
    cin >> i;
    cout << "Enter the numbers to be evaluated: " << endl;
    while (nums.size() < i && cin >> n){
        nums.push_back(n);
    }
    int target;
    cout << "Enter the target you want to search\n";
    cin >> target;
    int result = binarySearch(nums,target);
    if (result == -1){
        cout << "The target is not found ! ";
    }
    else{
        cout << "The target is present at index: " << result;
    }
    return 0;
}
